src/driver/mysql/MysqlDriver.ts

```diff
@@ -825,8 +829,10 @@ export class MysqlDriver implements Driver {
 
         /**
          * fix https://github.com/typeorm/typeorm/issues/1139
+         * note that if the db did support uuid column type it wouldn't have been defaulted to varchar
          */
-        if (column.generationStrategy === "uuid") return "36"
+        if (column.generationStrategy === "uuid" && column.type !== "uuid")
```

Comment:
correct me if I'm wrong, but from what I see for `mysql` generationStrategy is `uuid` and column type is `uuid` as well. But we don't return `36` in such case now, which is different from what we had before? If yes, this can cause issues for those who is using mysql right now.

Reply:
that's correct I believe. It looks like before if the generationStrategy was "uuid" it would default the column type from "uuid" to "varchar". Then here if no length was attributed to the column, it would plug in "36" since mysql needs a length for the varchar.

Yes, in returning I see your point this does cause issues for existing users that need the column type to remain as varchar given that it use to be defaulted to varchar. I'll have to rethink this one. Let me know if you have any suggestions.

Reply:
@pleerock my thoughts are that maybe PrimaryGeneratedColumnUUIDOptions should probably add one more property that allows for configuring the actual column type to be used so that this logic isn't tied up in the driver itself.

Reply:
Better to save the scope of this PR only to mariadb. So we just need to add proper checks I avoid any breaking changes for mysql users.

Reply:
yes good point.

For this, maybe it makes more sense to do this conditional to make it clear of the exact case its needed:
`
if (column.generationStrategy === "uuid" && column.type === "varchar") return 36
`

the reason its needed is because the generation strategy being uuid would default to varchar column type when normalized. Because of that when we get the column length it would wrongly do a length of 255 by default, but should be 36 in this case.

Other than that at line 727 the else if will check if the column type is "uuid" and if its a mysql driver it will default to varchar.

Let me know if that sounds correct and I can make the updates.

Reply:
yes, sounds good. But please check again everything before it will be merged, just to make sure we didn't break user use cases.

Reply:
added a private property for seeing if the uuid data type can be used. Also modified the conditional as mentioned. Added a regression test for mysql to the test suite. Let me know if anything else is needed.

Reply:
Could we get a merge? @smith-xyz

Reply:
Could we get a merge? @pleerock